src/app/clusters/mode-select-server/mode-select-server.h

```diff
@@ -0,0 +1,103 @@
+/*
+*
+*    Copyright (c) 2023 Project CHIP Authors
+*    All rights reserved.
+*
+*    Licensed under the Apache License, Version 2.0 (the "License");
+*    you may not use this file except in compliance with the License.
+*    You may obtain a copy of the License at
+*
+*        http://www.apache.org/licenses/LICENSE-2.0
+*
+*    Unless required by applicable law or agreed to in writing, software
+*    distributed under the License is distributed on an "AS IS" BASIS,
+*    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*    See the License for the specific language governing permissions and
+*    limitations under the License.
+*/
+
+#pragma once
+
+#include <app/AttributeAccessInterface.h>
+#include <app/clusters/mode-select-server/mode-select-delegate.h>
+#include <app/util/af.h>
+#include <map>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace ModeSelect {
+
+// todo have this list be generated by zap.
+const std::array<ClusterId , 6> AliasedClusters = {0x50, 0x51, 0x52, 0x54, 0x55, 0x59};
+
+class Instance : public CommandHandlerInterface, public AttributeAccessInterface
+{
+
+public:
+    // This map holds pointers to all initialised ModeSelect instances. It provides a way to access all ModeSelect clusters.
+    static std::map<uint32_t, Instance*> ModeSelectAliasesInstanceMap;
+
+    CHIP_ERROR Init();
+
+    // CommandHandlerInterface
+    void InvokeCommand(HandlerContext & ctx) override;
+    CHIP_ERROR EnumerateAcceptedCommands(const ConcreteClusterPath & cluster, CommandIdCallback callback, void * context) override;
+
+    // Checks if this instance supports a feature
+    bool HasFeature(ModeSelectFeature feature) const;
+
+    // AttributeAccessInterface
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+    CHIP_ERROR Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder) override;
+
+    // Generic accessor functions
+    EmberAfStatus GetFeature(uint32_t * value) const;
+    EmberAfStatus SetFeatureMap(uint32_t value) const;
+    EmberAfStatus GetCurrentMode(uint8_t *) const;
+    EmberAfStatus SetCurrentMode(uint8_t) const;
+    EmberAfStatus GetStartUpMode(DataModel::Nullable<uint8_t> &) const;
+    EmberAfStatus SetStartUpMode(uint8_t value) const;
+    EmberAfStatus SetStartUpModeNull() const;
+    EmberAfStatus GetOnMode(DataModel::Nullable<uint8_t> &) const;
+    EmberAfStatus SetOnMode(uint8_t value) const;
+    EmberAfStatus SetOnModeNull() const;
```

Comment:
I assume these have the form they do because these are being stored in the attribute store?  But if we have an attribute access interface anyway, maybe the right answer is to not store them there?  I guess the complication is handling persistence to NVM for CurrentMode, StartUpMode, OnMode?  But it might actually be a bit less annoying to use `app::GetAttributePersistenceProvider()` directly if we just want to do that.... and I guess manually marking things dirty on write. Worth thinking about, at least.